### PR TITLE
Add replace/create/delete to the generic client.

### PR DESCRIPTION
**Pull request by brendandburns**
Builds on #24

### CODE REVIEW

**Comment by ityuhui** [comment]
Thank you @brendandburns 

It's great that kubernetes-c/client has a generic client now.

Only a small update:

We'd better integrate the build check of examples/generic to Github/Action
in https://github.com/kubernetes-client/c/blob/master/examples/Makefile

```makefile
all：
	cd generic; make

clean:
	cd generic; make clean
```

File: kubernetes/include/generic.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#ifndef __GENERIC_H__
+#define __GENERIC_H__
+
```
Comment (ityuhui) [inline kubernetes/include/generic.h:3]: Could you please add the macro for c++ support ?

```c
#ifdef  __cplusplus
extern "C" {
#endif

#ifdef  __cplusplus
}
#endif
```

File: kubernetes/CMakeLists.txt [hunk 2]
```diff
@@ -41,6 +41,7 @@ set(SRCS
     src/apiKey.c
     src/apiClient.c
     src/binary.c
+    src/generic.c
     external/cJSON.c
     model/object.c
     model/admissionregistration_v1_service_reference.c
```
Comment (ityuhui) [inline kubernetes/CMakeLists.txt:47]: Maybe kubernetes/include/generic.h is needed to added to 

```
set(HDRS
    include/generic.h
```
in kubernetes/CMakeLists.txt ?

File: kubernetes/include/generic.h (hunk 1 above)
Comment (brendandburns) [reply kubernetes/include/generic.h:3]: done.

File: kubernetes/CMakeLists.txt (hunk 2 above)
Comment (brendandburns) [reply kubernetes/CMakeLists.txt]: done.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

**Comment by ityuhui** [comment]
/lgtm
/approve

Thank you for creating the wonderful generic client !

**Pull request merged**